crates/rome_formatter/src/ts/patterns/assign_pattern.rs

```diff
@@ -1,14 +1,16 @@
-use crate::{empty_element, format_elements, FormatElement, Formatter, ToFormatElement};
+use crate::{
+	empty_element, format_elements, FormatElement, FormatError, Formatter, ToFormatElement,
+};
 use rslint_parser::ast::AssignPattern;
 
 impl ToFormatElement for AssignPattern {
-	fn to_format_element(&self, formatter: &Formatter) -> Option<FormatElement> {
-		let key = if let Some(key) = self.key() {
+	fn to_format_element(&self, formatter: &Formatter) -> Result<FormatElement, FormatError> {
+		let key = if let Ok(key) = self.key() {
 			formatter.format_node(key)?
 		} else {
 			empty_element()
 		};
-		let assign = if let Some(eq_token) = self.eq_token() {
+		let assign = if let Ok(eq_token) = self.eq_token() {
 			formatter.format_token(&eq_token)?
 		} else if let Some(colon_token) = self.colon_token() {
 			formatter.format_token(&colon_token)?
```

Comment:
There are probably more cases but we shouldn't format a missing token. We should just bail out if a mandatory token is missing. This is true in all formatting code (I only just realised this now, there are probably more such examples in other files).

Or is this not a mandatory token in which case the grammar is incorrect? It might then be worth adding a TODO (pointing to #1725) or explain in the PR summary that you intentionally didn't change the grammar.

```suggestion
		let assign = formatter.format_token(&eq_token?)?
```